### PR TITLE
Allow to enable RBAC

### PR DESCRIPTION
**Pull request by mcanevet**

### CODE REVIEW

**Comment by yupwei68** [comment]
Hi @mcanevet , thanks for opening this PR.  
As https://github.com/Azure/terraform-azurerm-aks/pull/53, 1. Please keep the unrelated changed. 2. Please add this feature into fixture/test and test if it works fine. 3. Please add this feature into the example of README.

**Comment by mcanevet** [comment]
@yupwei68 done

**Comment by mcanevet** [comment]
@yupwei68 rebased and fixed

**Comment by mcanevet** [comment]
@yupwei68 is there something that prevents this to be merged?

**Comment by yupwei68** [comment]
@mcanevet The test has failed, look like that I don't have permission to create the group. I'll  test it using a local service principal, which might be a more general case later when I'm available.

File: main.tf [hunk 1]
```diff
@@ -67,6 +67,14 @@ resource "azurerm_kubernetes_cluster" "main" {
     }
   }
 
+  role_based_access_control {
+    enabled = length(var.admin_group_object_ids) > 0
```
Comment (derkoe) [inline main.tf:71]: Would be good to set this with a separate variable

File: main.tf [hunk 2]
```diff
@@ -67,6 +67,14 @@ resource "azurerm_kubernetes_cluster" "main" {
...
+    azure_active_directory {
```
Comment (derkoe) [inline main.tf:72]: This block should be added with the `admin_group_object_ids` set

File: variables.tf [hunk 3]
```diff
@@ -97,3 +97,9 @@ variable "sku_tier" {
   type        = string
   default     = "Free"
 }
+
+variable "admin_group_object_ids" {
```
Comment (derkoe) [inline variables.tf:101]: Maybe rename this to `aad_admin_group_object_ids`

File: main.tf (hunk 2 above)
Comment (mcanevet) [reply main.tf:72]: @derkoe I'm not sure to understand what you mean.

**Comment by mcanevet** [comment]
@yupwei68 is there anything I can do to having this merged?
This is the only blocker for us to use this module.

**Comment by derkoe** [comment]
@mcanevet  BTW our company also hosts an AKS module with some more options:  https://registry.terraform.io/modules/porscheinformatik/aks/azurerm/latest

**Comment by mcanevet** [comment]
@derkoe I saw that, but I prefer to contribute and use Azure's one, hoping that one day it will fit my needs.

**Comment by yupwei68** [comment]
@mcanevet Sorry that I'm currently occupied with other works with higher priority. Thanks for you contribution! 🚀  I'll review it before the mid of Dec. Thanks!

File: main.tf [hunk 4]
```diff
@@ -67,6 +67,14 @@ resource "azurerm_kubernetes_cluster" "main" {
     }
   }
 
+  role_based_access_control {
+    enabled = var.enable_role_based_access_control
+    azure_active_directory {
+      managed                = true
+      admin_group_object_ids = var.aad_admin_group_object_ids
```
Comment (yupwei68) [inline main.tf:74]: Would you add another option when `managed` is `false`, and `client_app_id`, `server_app_id` and `server_app_secret` is set?

**Review by yupwei68 — COMMENTED** [review]
Thanks for your update. Except this change, would you paste the test result by the following cmd if you're convenient (for the Administrator permission, I may need test it later):
`docker build --build-arg BUILD_ARM_SUBSCRIPTION_ID=$ARM_SUBSCRIPTION_ID --build-arg BUILD_ARM_CLIENT_ID=$ARM_CLIENT_ID --build-arg BUILD_ARM_CLIENT_SECRET=$ARM_CLIENT_SECRET --build-arg BUILD_ARM_TENANT_ID=$ARM_TENANT_ID -t azure-aks .`
`docker run --rm azure-aks /bin/bash -c "bundle install && rake full"`

File: main.tf (hunk 4 above)
Comment (mcanevet) [reply main.tf]: Done

**Comment by mcanevet** [comment]
@yupwei68 here the output of the tests : https://gist.github.com/mcanevet/b261af8bf9c0b9cbb6deb6d09762db7d

**Review by yupwei68 — COMMENTED** [review]
Thanks again for your contribution 🚀 . This PR looks almost good to me. Sorry that we'll need to reverse the changes in test/fixture cause my testing sp lacks permissions. Except these changes, I think the PR is ok.

File: README.md [hunk 5]
```diff
@@ -26,16 +26,21 @@ module "network" {
   depends_on          = [azurerm_resource_group.example]
 }
 
+resource "azuread_group" "aks_cluster_admins" {
+  name = "AKS-cluster-admins"
+}
+
```
[the remark this answers is not in the document]
Comment (mcanevet) [reply README.md]: Is it OK with a datasource instead of a resource?

[the remark this answers is not in the document]
Comment (yupwei68) [reply README.md]: Datasource needs graph permission to list all groups in the subs as well. We could just specify the object id of an AAD group.

**Comment by mcanevet** [comment]
@yupwei68 is it OK now?

**Review by yupwei68 — APPROVED** [review]
LGTM!

**Comment by mcanevet** [comment]
@yupwei68 Thanks a lot. Would it be possible to release the new version so that we can point to it?

**Comment by yupwei68** [comment]
@mcanevet I've released in v4.3.0.

**Pull request merged**